ToolChain/Assembler/Parser/Parser.py

```diff
@@ -20,586 +19,412 @@
     4. Once all info has been evaluated, if correct, we generate the binary code for each instruction
     """
 
-    loadAddress = 0x00
-    instructionList = None
-    fileContent = None
-    referenceDict = None
-    finalSize = 0
+    relativeAddress = 0                       # Used to determine memory address of a label at a given point in the file
 
-    def __init__(self, file: str="", skipValidation=False):
+    def parse(self, text):
         """
-        The initial read of the file and parsing of the file is done when Parser is initialized.
-        This is a "one stop shop" to get the information easily and fast without requiring multiple
-        step. The point of the whole class, is, after all to get these information. No need for
-        multiple steps.
-        :param file: String, the name of the file that needs to be read in to build the code
-        :return: Nothing. Just the normal init stuff...
+        This takes in an individual line of code, parses the info and returns the fully built binary code, or returns a
+        memory reference with the appropriate offset.
+        :param text: Str, Individual line of code.
+        :return: tuple: Represents the binary instruction, the current offset, and the appropriate flag
         """
 
-        if file is None or type(file) is not str or len(file) <= 1 and not skipValidation:
-            raise ValueError("Parser error - file has to be defined")
+        self.relativeAddress = 0                # Reset offset (instruction size) for every line parsed
+        line = text.split()                     # We split the line with space as a delimiter. Each token is evaluated
+        instruction = b''                       # Instruction will be written as a bytestring
+        operandList = []                        # We keep each operand in a list to evaluate later
+        form = "Ins"                            # We assume the first token of the line is an instruction
+        labelFlag = 0                           # Used by the assembler to determine what we're returning
+
+        # This ensures we don't evaluate an empty line. This also handles a line full of spaces
+        if len(line) == 0:
+            return "", self.relativeAddress, 3
+
+        # First off, we need to determine if this line has a label or global label to be referenced
+        # If so, we can simply return to the Assembler class with the label, the offset, and the appropriate flag
+        identifierMatch = re.search(r'(\.(\w*))|(\w*:)|(;)', line[0].upper())
+
+        if identifierMatch:
+            # We had a match for identifier patterns, so we assume there's no instruction.
+            instruction, labelFlag = self._evaluateIndicatorData(text, line[0].upper(), line, instruction)
+            return instruction, self.relativeAddress, labelFlag
+
+        elif line[0].upper() in INSTRUCTION_LIST:
```

Comment:
I still think we should use this:

```python3
line = line.upper().split(';', maxsplit=1)[0]
fragments = line.split()
if re.search(pattern, line):
   # handle line
   ret
elif fragments[0] in INSTRUCTION_LIST:
   # do rest
   ret
else:
   # err
```

Reply:
I'm all in on using ";" for a first split.